src/common/transformations/src/transformations/common_optimizations/reshape_sequence_fusion.cpp

```diff
@@ -17,21 +17,53 @@
 NGRAPH_RTTI_DEFINITION(ngraph::pass::ReshapeSequenceFusion, "ReshapeSequenceFusion", 0);
 
 namespace {
+OPENVINO_SUPPRESS_DEPRECATED_START
 bool has_valid_pattern(const std::shared_ptr<ngraph::Node> & node) {
     auto const_node = std::dynamic_pointer_cast<ngraph::opset8::Constant>(node);
-    if (!const_node) return false;
+    if (!const_node) {
+        // Lower bound of the value
+        ngraph::HostTensorPtr lb = std::make_shared<ngraph::HostTensor>(node->output(0));
+        const bool is_lb_evaluated = node->evaluate_lower({lb});
+        if (!is_lb_evaluated || !lb) return false;
+        const auto lb_const_node = std::make_shared<ngraph::opset8::Constant>(lb);
+        if (!lb_const_node) return false;
+
+        const auto & lb_values = lb_const_node->cast_vector<int64_t>();
+        bool is_valid = std::all_of(lb_values.cbegin(), lb_values.cend(), [](int64_t value) { return value > 0;});
+        if (is_valid) {
+            return true;
+        }
+
+        // Upper bound of the value
+        ngraph::HostTensorPtr ub = std::make_shared<ngraph::HostTensor>(node->output(0));
+        const bool is_ub_evaluated = node->evaluate_upper({ub});
+        if (!is_ub_evaluated || !ub) return false;
+        const auto ub_const_node = std::make_shared<ngraph::opset8::Constant>(ub);
+        if (!ub_const_node) return false;
+
+        const auto & ub_values = ub_const_node->cast_vector<int64_t>();
+        if (lb_values.size() != ub_values.size()) return false;
+
+        is_valid = std::all_of(lb_values.cbegin(), lb_values.cend(), [](int64_t value) { return value == 0;});
```

Comment:
What about the case when lb = {0, 1, 1} and ub = {max_int, 64, 64} ? in this case I guess we return false, but I think we should we here just first dimension for pure dynamism and skip others since they > 0.

Reply:
You are right and the `ReshapeSequenceFusionNeg6` with `ShapeOf({-1, 3, -1})` test was about to check that, as I thought we want every value to be fully dynamic. To accept mixed option I've relaxed the checks, and the corresponding test is now `ReshapeSequenceFusion5`.